src/main/java/org/opensearch/ad/rest/RestDeleteAnomalyResultsAction.java

```diff
@@ -70,21 +88,32 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         DeleteByQueryRequest deleteRequest = new DeleteByQueryRequest(ALL_AD_RESULTS_INDEX_PATTERN)
             .setQuery(searchSourceBuilder.query())
             .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_HIDDEN);
-        return channel -> client.execute(DeleteAnomalyResultsAction.INSTANCE, deleteRequest, ActionListener.wrap(r -> {
-            XContentBuilder contentBuilder = r.toXContent(channel.newBuilder().startObject(), ToXContent.EMPTY_PARAMS);
-            contentBuilder.endObject();
-            channel.sendResponse(new BytesRestResponse(RestStatus.OK, contentBuilder));
-        }, e -> {
-            try {
-                channel.sendResponse(new BytesRestResponse(channel, e));
-            } catch (IOException exception) {
-                logger.error("Failed to send back delete anomaly result exception result", exception);
-            }
-        }));
+        CompletableFuture<BulkByScrollResponse> futureResponse = new CompletableFuture<>();
+        client
+            .execute(
+                DeleteAnomalyResultsAction.INSTANCE,
+                deleteRequest,
+                ActionListener
+                    .wrap(deleteResponse -> futureResponse.complete(deleteResponse), ex -> futureResponse.completeExceptionally(ex))
+            );
+        BulkByScrollResponse deleteByQueryResponse = futureResponse
+            .orTimeout(
+                AnomalyDetectorSettings.REQUEST_TIMEOUT.get(extensionsRunner.getEnvironmentSettings()).getMillis(),
+                TimeUnit.MILLISECONDS
+            )
+            .join();
+
+        XContentBuilder contentBuilder = deleteByQueryResponse
+            .toXContent(JsonXContent.contentBuilder().startObject(), ToXContent.EMPTY_PARAMS);
```

Comment:
👍

Reply:
Just a thought (not for this PR).  I keep seeing this boilerplate.

Somewhere (possibly BaseExtensionRestHandler?) we should declare a constant:
```java
imprt org.opensearch.core.xcontent.ToXContent;

public static XContentBuilder EMPTY_JSON_XCONTENT = (new ToXContent() {
    @Override  
    toXContent(XContentBuilder builder, Params params) throws IOException {
        builder.endObject();
        return builder;
    }).toXContent(JsonXContent.contentBuilder().startObject(), ToXContent.EMPTY_PARAMS);
}
```

Reply:
That's a good point but the above contentBuilder code is specific to Delete Detector Results since it has BulkScrollResponse which needs `startObject` and `endObject`. The common code we have for the other rest actions is:

```java
  return new ExtensionRestResponse(
            request,
            RestStatus.OK,
            response.toXContent(JsonXContent.contentBuilder(), ToXContent.EMPTY_PARAMS)
        );
```
Do you think we should add `response.toXContent(JsonXContent.contentBuilder(), ToXContent.EMPTY_PARAMS)` to `BaseExtensionRestHandler`?